### PR TITLE
FIX Jenkins 39809 404 pages appear only in pipeline page children

### PR DESCRIPTION
**Pull request by scherler**
# Description

See [JENKINS-39809](https://issues.jenkins-ci.org/browse/JENKINS-39809).

I left a FIXME note for @imeredith since I am not yet familiar with the mobx stuff and we need to return some object that indicates that we have to show the 404.

AT PR: https://github.com/jenkinsci/blueocean-acceptance-test/pull/71
Passing ATH: https://ci.blueocean.io/blue/organizations/jenkins/ATH-Jenkinsfile/detail/JENKINS-39809_404_pages_appear_only_in_pipelinePage_children/4

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [x] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [x] Run the changes and verified the change matches the issue description
- [x] Reviewed the code
- [x] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
Some things I tried:

Activity screen with non existing pipeline name: works
Dashboard with nonsense path: works
Result screen with non existent pipeline name or run: doesn't work, although API shows up 404. 

So if the scope of this is narrowed to be those - then :bee: from be (and thanks for ATH too!)

Can you make a follow on ticket for Ivan or something to deal with the results screen not showing 404 @scherler ?

File: blueocean-core-js/src/js/services/ActivityService.js [hunk 1]
```diff
@@ -90,6 +90,9 @@ export class ActivityService extends BunkerService {
                     run.result = 'UNKNOWN';
                 }
                 return this.setItem(run);
+            })
+            .catch(err => {
+                console.log('There had been an error while trying to get the data.', err); // FIXME: Ivan what is the way to return an "error" opbject so underlying component are aware of the problem and can react
```
Comment (michaelneale) [inline blueocean-core-js/src/js/services/ActivityService.js]: "There has" - as past tense doesn't make sense here :ant:

File: blueocean-dashboard/package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
   },
   "devDependencies": {
     "@jenkins-cd/eslint-config-jenkins": "0.0.2",
-    "@jenkins-cd/js-builder": "0.0.47",
+    "@jenkins-cd/js-builder": "0.0.49-beta6",
```
Comment (michaelneale) [inline blueocean-dashboard/package.json]: I wonder what goodness this brings

Comment (scherler) [inline blueocean-dashboard/package.json]: That is fixing a bug that we discovered when I moved the NotFound to core and had to introduce `react-router` as dep. Without that version that leads to an exception for the bundle. Kudos to @tfennelly

File: blueocean-core-js/src/js/services/ActivityService.js (hunk 1 above)
Comment (scherler) [inline blueocean-core-js/src/js/services/ActivityService.js]: I talked to @imeredith and he wanted to extend the pager to return an object such as:

```
{ data, error, state }
```

That will hopefully go away anyway but will fix the working.

**Comment by scherler** [comment]
@michaelneale if you want we keep this PR open.

however I personally try to get PRs quickly in (still remembering the endless i18n branch which costs extra time to keep it in sync with master.

**Comment by scherler** [comment]
https://issues.jenkins-ci.org/browse/JENKINS-40032 is the followup for @imeredith

**Pull request merged**